tests/acceptance/features/bootstrap/DataExporterContext.php

```diff
@@ -256,14 +280,52 @@ private function assertFilePhysicallyExistInLastExport($filename, $message = '')
 	/**
 	 * @param string $filename
 	 * @param string $content
+	 * @param bool $trash
 	 *
 	 * @return void
 	 */
-	private function assertFilePhysicallyExistInLastExportWithContent($filename, $content) {
-		$this->featureContext->theFileWithContentShouldExistInTheServerRoot(
-			self::path("$this->lastExportPath/files/$filename"),
-			$content
-		);
+	private function assertFilePhysicallyExistInLastExportWithContent($filename, $content, $trash = false) {
+		if ($trash) {
+			$this->featureContext->listTrashbinFileInServerRoot(self::path("$this->lastExportPath/files_trashbin"));
+			$matches = [];
+			foreach (HttpRequestHelper::getResponseXml($this->featureContext->getResponse())->data->element as $item) {
+				$file = (string)$item;
+				if (\strpos($file, $filename) === 0) {
+					$matches[] = $file;
+				}
+			}
+			$fileFound = false;
+			$fileCount = \count($matches);
+			foreach ($matches as $match) {
+				$path = "$this->lastExportPath/files_trashbin/$match";
+				$this->featureContext->readFileInServerRoot($path);
+				Assert::assertSame(
+					200,
+					$this->featureContext->getResponse()->getStatusCode(),
+					"Failed to read the file {$path}"
+				);
+				$fileContent = HttpRequestHelper::getResponseXml($this->featureContext->getResponse());
+				$fileContent = (string)$fileContent->data->element->contentUrlEncoded;
+				$fileContent = \urldecode($fileContent);
+
+				while ($fileCount > 0) {
+					$fileCount--;
+					if ($fileContent === $content) {
+						$fileFound = true;
+						break 2;
+					} elseif ($fileCount > 0 && $fileFound == false) {
+						break;
+					} else {
+						Assert::fail("The content of the file does not match with '{$content}'");
+					}
+				}
+			}
```

Comment:
I dont really understand the logic here. I think you can replace it with something simple like
``` php
$fileFound = false;
foreach ($matches as $match) {
	$path = "$this->lastExportPath/files_trashbin/$match";
	$this->featureContext->readFileInServerRoot($path);
	$fileContent = HttpRequestHelper::getResponseXml($this->featureContext->getResponse());
	$fileContent = (string)$fileContent->data->element->contentUrlEncoded;
	$fileContent = \urldecode($fileContent);
	if ($fileContent === $content) {
		$fileFound = true;
		break;
	}
}
if (!$fileFound) {
	Assert::fail("error msg.");
}
```

Reply:
I have used this logic for multiple files in trashbin. If there is a single file then as you have suggested the simple logic will work but for multiple files it might fail as @kiranparajuli589   has mentioned earlier.

Reply:
> I have used this logic for multiple files in trashbin. If there is a single file then as you have suggested the simple logic will work but for multiple files it might fail as @kiranparajuli589 has mentioned earlier.

for multiple files, just add the above logic with a for loop of filenames like:
``` php
foreach($paths as $path) {
    $fileFound = false;
    foreach ($matches as $match) {
	$path = "$this->lastExportPath/files_trashbin/$match";
	$this->featureContext->readFileInServerRoot($path);
	$fileContent = HttpRequestHelper::getResponseXml($this->featureContext->getResponse());
	$fileContent = (string)$fileContent->data->element->contentUrlEncoded;
	$fileContent = \urldecode($fileContent);
	if ($fileContent === $content) {
		$fileFound = true;
		break;
	}
    }
    if (!$fileFound) {
	Assert::fail("error msg.");
    }
}

Reply:
Yes, the code mentioned above also works for multiple files. We are loopint through all the trashbin files in the for loop and checking the content of each file.

Reply:
Done